lib/auth/vnetconfig/v1/service.go

```diff
@@ -0,0 +1,201 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package vnetconfig
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/gravitational/teleport/api/gen/proto/go/teleport/vnet/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/services/local"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+type Service struct {
+	// Opting out of forward compatibility, this service must implement all service methods.
+	vnet.UnsafeVnetConfigServiceServer
+
+	storage    *local.VnetConfigService
+	authorizer authz.Authorizer
+}
+
+func NewService(storage *local.VnetConfigService, authorizer authz.Authorizer) *Service {
+	return &Service{
+		storage:    storage,
+		authorizer: authorizer,
+	}
+}
+
+func (s *Service) GetVnetConfig(ctx context.Context, _ *vnet.GetVnetConfigRequest) (*vnet.VnetConfig, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authCtx.CheckAccessToKind(types.KindVnetConfig, types.VerbRead); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	vnetConfig, err := s.storage.GetVnetConfig(ctx)
+	if err != nil {
+		return nil, utils.OpaqueAccessDenied(err)
```

Comment:
I don't like this function. It's called `OpaqueAccessDenied` but it also works with trace.Notfound errors :cry: 
It's really confusing how would storage layer complain about permissions 

Besides, there is inconsistency between this and other calls bellow like in update where the storage layer error is returned

Reply:
It should be called `OpaqueNotFound` since that's what it returns, but i don't want to go to the effort to change it right now

but to do this right you do need to return the exact same error for NotFound and the access check

in update i don't think we should obscure any errors, if you have update verb you should be allowed to know if an existing resource exists or not

Reply:
took it out because it doesn't matter without `where` support, you either have `read` verb and you're allowed to know if it exists, or you don't and you get access denied